### ISSUE TITLE
Use python_requires to prevent installation on unsupported Python versions

### ISSUE THREAD

**Issue by ThreePinkApples** [crossbario/autobahn-python#1293]
With the current newest release, 20.1.1, support for Python 2 broke (as expected). But a Python 2 environment will still install 20.1.1 unless it is prevented through requirements.
A better, and global, solution is to set `python_requires` in setup.py, so that pip and other package managers won't try to install unsupported versions, but will pick the older one (19.11.2) that does support Python 2

https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires

Although for this to take proper effect you'd need to republish 20.1.1 (or delete it) with this change

**Comment by oberstet** [comment]
yep, agreed, we should totally add that (for the next release)

> note: we did change the classifiers for py3-only https://github.com/crossbario/autobahn-python/blob/master/setup.py#L272 .. but forget to add `python_requires` (which we didn't need up till now since autobahn supported py2+3)

**Comment by oberstet** [comment]
ok, gonna release 20.1.2 as a hot fix via https://github.com/crossbario/autobahn-python/pull/1294 and delete 20.1.1

works as expected locally (I am gonna test pip install on py2 after the release too):

```
oberstet@intel-nuci7:~$ ~/cpy27/bin/pip install virtualenv
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Requirement already satisfied: virtualenv in ./.local/lib/python2.7/site-packages (16.6.0)
oberstet@intel-nuci7:~$ ~/cpy27/bin/python -m venv ~/cpy27_1
/home/oberstet/cpy27/bin/python: No module named venv
oberstet@intel-nuci7:~$ ~/cpy27/bin/python -m virtualenv ~/cpy27_1
New python executable in /home/oberstet/cpy27_1/bin/python
Installing setuptools, pip, wheel...
done.
oberstet@intel-nuci7:~$ source ~/cpy27_1/bin/activate
(cpy27_1) oberstet@intel-nuci7:~$ cd scm/crossbario/autobahn-python/
(cpy27_1) oberstet@intel-nuci7:~/scm/crossbario/autobahn-python$ make install 
pip uninstall -y pytest_asyncio # remove the broken shit
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
WARNING: Skipping pytest-asyncio as it is not installed.
pip uninstall -y pytest_cov # remove the broken shit
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
WARNING: Skipping pytest-cov as it is not installed.
# enforce use of bundled libsodium
AUTOBAHN_USE_NVX=1 SODIUM_INSTALL=bundled pip install -e .[all]
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Obtaining file:///home/oberstet/scm/crossbario/autobahn-python
ERROR: Package 'autobahn' requires a different Python: 2.7.17 not in '>=3.5'
Makefile:15: recipe for target 'install' failed
make: *** [install] Error 1
(cpy27_1) oberstet@intel-nuci7:~/scm/crossbario/autobahn-python$ which python
/home/oberstet/cpy27_1/bin/python
(cpy27_1) oberstet@intel-nuci7:~/scm/crossbario/autobahn-python$ python -V
Python 2.7.17
(cpy27_1) oberstet@intel-nuci7:~/scm/crossbario/autobahn-python$ 
```

**Comment by oberstet** [comment]
ok, released v20.1.2 with above hotfix:

* https://pypi.org/project/autobahn/20.1.2/
* https://github.com/crossbario/autobahn-python/tree/v20.1.2

also, deleted the 20.1.1 package on pypi (but not the tag here). seems to work (auto-select the right version) .. tested in fresh venvs:

# py2/pip

```
(cpy27_2) oberstet@intel-nuci7:~$ ~/cpy27/bin/python -m virtualenv ~/cpy27_3
New python executable in /home/oberstet/cpy27_3/bin/python
Installing setuptools, pip, wheel...
done.
(cpy27_2) oberstet@intel-nuci7:~$ source ~/cpy27_3/bin/activate
(cpy27_3) oberstet@intel-nuci7:~$ python -c "import autobahn; print(autobahn.version)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named autobahn
(cpy27_3) oberstet@intel-nuci7:~$ pip install --no-cache autobahn
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Collecting autobahn
  Downloading https://files.pythonhosted.org/packages/84/05/828b0c5a558b67c8afd64e233976553298688eee39c78de3a8c8cf98c91f/autobahn-19.11.2-py2.py3-none-any.whl (771kB)
     |████████████████████████████████| 778kB 1.7MB/s 
Collecting six>=1.11.0
  Downloading https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting txaio>=18.8.1
  Downloading https://files.pythonhosted.org/packages/e9/6d/e1a6f7835cde86728e5bb1f577be9b2d7d273fdb33c286e70b087d418ded/txaio-18.8.1-py2.py3-none-any.whl
Collecting cryptography>=2.7
  Downloading https://files.pythonhosted.org/packages/06/1c/957cd27e4467b21e22b8bfa3714193cf4669ebece1800477cafba4321e34/cryptography-2.8-cp27-cp27m-manylinux2010_x86_64.whl (2.3MB)
     |████████████████████████████████| 2.3MB 7.8MB/s 
Collecting enum34; python_version < "3"
  Downloading https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting ipaddress; python_version < "3"
  Downloading https://files.pythonhosted.org/packages/c2/f8/49697181b1651d8347d24c095ce46c7346c37335ddc7d255833e7cde674d/ipaddress-1.0.23-py2.py3-none-any.whl
Collecting cffi!=1.11.3,>=1.8
  Downloading https://files.pythonhosted.org/packages/d0/d6/fd4d410d3429b0e93bf0fb80fd59a7ae4663ca0c8440e5d95142beb86e7f/cffi-1.13.2-cp27-cp27m-manylinux1_x86_64.whl (385kB)
     |████████████████████████████████| 389kB 14.3MB/s 
Collecting pycparser
  Downloading https://files.pythonhosted.org/packages/68/9e/49196946aee219aead1290e00d1e7fdeab8567783e83e1b9ab5585e6206a/pycparser-2.19.tar.gz (158kB)
     |████████████████████████████████| 163kB 14.2MB/s 
Building wheels for collected packages: pycparser
  Building wheel for pycparser (setup.py) ... done
  Created wheel for pycparser: filename=pycparser-2.19-py2.py3-none-any.whl size=111027 sha256=70ced8a88b05a6d6dabc1d625673a6d1c477d7fd343f0a67d510d96321a7e1e0
  Stored in directory: /tmp/pip-ephem-wheel-cache-I7a34g/wheels/f2/9a/90/de94f8556265ddc9d9c8b271b0f63e57b26fb1d67a45564511
Successfully built pycparser
Installing collected packages: six, txaio, enum34, ipaddress, pycparser, cffi, cryptography, autobahn
Successfully installed autobahn-19.11.2 cffi-1.13.2 cryptography-2.8 enum34-1.1.6 ipaddress-1.0.23 pycparser-2.19 six-1.13.0 txaio-18.8.1
(cpy27_3) oberstet@intel-nuci7:~$ python -c "import autobahn; print(autobahn.version)"
19.11.2
(cpy27_3) oberstet@intel-nuci7:~$ 
```

# py3/pip

```
(cpy381_1) oberstet@intel-nuci7:~$ ~/cpy381/bin/python3 -m venv ~/cpy381_2
(cpy381_1) oberstet@intel-nuci7:~$ source ~/cpy381_2/bin/activate
(cpy381_2) oberstet@intel-nuci7:~$ python -c "import autobahn; print(autobahn.version)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'autobahn'
(cpy381_2) oberstet@intel-nuci7:~$ pip install --no-cache autobahn
Collecting autobahn
  Downloading https://files.pythonhosted.org/packages/b4/db/f9762ed48eeae8ca2c30c9313b8f0fc05843925ddf578db89bb67c9e1f44/autobahn-20.1.2-py2.py3-none-any.whl (767kB)
     |████████████████████████████████| 768kB 1.8MB/s 
Collecting cryptography>=2.7 (from autobahn)
  Downloading https://files.pythonhosted.org/packages/ca/9a/7cece52c46546e214e10811b36b2da52ce1ea7fa203203a629b8dfadad53/cryptography-2.8-cp34-abi3-manylinux2010_x86_64.whl (2.3MB)
     |████████████████████████████████| 2.3MB 14.5MB/s 
Collecting six>=1.11.0 (from autobahn)
  Downloading https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting txaio>=18.8.1 (from autobahn)
  Downloading https://files.pythonhosted.org/packages/e9/6d/e1a6f7835cde86728e5bb1f577be9b2d7d273fdb33c286e70b087d418ded/txaio-18.8.1-py2.py3-none-any.whl
Collecting cffi!=1.11.3,>=1.8 (from cryptography>=2.7->autobahn)
  Downloading https://files.pythonhosted.org/packages/05/7d/723ceca757d8016850d1110df44b91121695f22f782f876b483ecb3928a9/cffi-1.13.2-cp38-cp38-manylinux1_x86_64.whl (439kB)
     |████████████████████████████████| 440kB 15.0MB/s 
Collecting pycparser (from cffi!=1.11.3,>=1.8->cryptography>=2.7->autobahn)
  Downloading https://files.pythonhosted.org/packages/68/9e/49196946aee219aead1290e00d1e7fdeab8567783e83e1b9ab5585e6206a/pycparser-2.19.tar.gz (158kB)
     |████████████████████████████████| 163kB 14.5MB/s 
Installing collected packages: pycparser, cffi, six, cryptography, txaio, autobahn
  Running setup.py install for pycparser ... done
Successfully installed autobahn-20.1.2 cffi-1.13.2 cryptography-2.8 pycparser-2.19 six-1.13.0 txaio-18.8.1
WARNING: You are using pip version 19.2.3, however version 19.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
(cpy381_2) oberstet@intel-nuci7:~$ python -c "import autobahn; print(autobahn.version)"
20.1.2
(cpy381_2) oberstet@intel-nuci7:~$ 
```
